service/history/workflow/update/validation.go

```diff
@@ -85,6 +85,9 @@ func validateRequestMsgPrefix(
 func validateAcceptanceMsg(msg *updatepb.Acceptance) error {
 	return validate(
 		notZero(msg, "body", msg),
+		notZero(msg.GetAcceptedRequestSequencingEventId(), "accepted_request_sequencing_event_id", msg),
+		notZero(msg.GetAcceptedRequestMessageId(), "accepted_request_message_id", msg),
+		notZero(msg.GetAcceptedRequest(), "accepted_request", msg),
```

Comment:
I don't think we should be demanding that the request is in the protocol message because the server doesn't take the request from the `Accepted` protocol message; it takes it from the in-memory update registry:

https://github.com/temporalio/temporal/blob/main/service/history/workflow/update/update.go#L415-L424 

Furthermore I don't think SDKs are sending the request payload in the Accepted protocol message. @Sushisource isn't this saying that SDKs send an empty Accepted message?

https://github.com/temporalio/sdk-core/blob/master/sdk-core-protos/protos/local/temporal/sdk/core/workflow_commands/workflow_commands.proto#L331

Reply:
Yes, Dan, you are right. What's about `AcceptedRequestMessageId`. It is not actually used on server and I don't even remember why we decided to have it.

Reply:
I don't think the core SDKs are sending any of them with the Accepted protocol message: just the protocol instance ID. E.g.

https://github.com/temporalio/sdk-typescript/blob/main/packages/workflow/src/internals.ts#L795

Reply:
If that's right then the features repo tests should fail... once we merge the update tests.

Reply:
It's because we're running the feature tests against a server without update support

```
910:2024-01-18T00:36:59.4993530Z INFO:harness.python.feature:Skipping feature update/basic because server support for update is disabled; set frontend.enableUpdateWorkflowExecution=true in dynamic config to enable
```

Reply:
@dandavison The thing you linked to is the proto between Core and Lang, it isn't what gets sent back to server.

What you want to be looking at is [here](https://github.com/temporalio/sdk-core/blob/ee7f6caaa9ad00bca2e2db8093d444ce563d6016/core/src/worker/workflow/machines/update_state_machine.rs#L237) where we can see both fields are assigned.

However, it was good you mention this, because the `accepted_request` field is not assigned (and, in fact, I'm not entirely sure what that's even for...). So, we definitely don't want to assert that that is always assigned.

Also it would make sense for the integ tests here to use the actual Go SDK to perform updates too, if that's not in here already somewhere.

Reply:
OK, thanks, that makes sense. Yes I sort of knew that `accepted_request` could not be sent because TS doesn't send it and I assumed sdk-core wouldn't hold on to request payloads itself. (Although I could also have simply put some print statements in the server `handleCommand()`).

>  `accepted_request` field is not assigned (and, in fact, I'm not entirely sure what that's even for...).

The `Accepted` event that's written to history needs the request payload. So I think at one point there was a plan to have it round-tripped via the worker; but currently its held in the per-shard in-memory update "registry", and we hope it's still there when the Acceptance comes in from the worker
 https://github.com/temporalio/temporal/blob/main/service/history/workflow/update/update.go#L415-L424

Reply:
If that works and we keep it that way (which seems preferable to me than having the potentially amnesiac worker store it) then we should remove the field at some point.

Reply:
There's another aspect of having the workflow update validator function mutate the update input, we didn't go that direction in the SDK but it has been requested by users.
It makes sense to me to keep this capability on the server.